sip/execution_control/configuration_db/redis/db_client/events.py

```diff
@@ -0,0 +1,329 @@
+# coding=utf-8
+"""Execution Control Configuration database events module.
+
+An aggregate is a domain-driven-design (DDD) concept.
+You apply a command to an aggregate which then produces one or more events.
+
+For SIP the aggregates are the:
+
+    - Processing Block
+    - Scheduling Block Instance
+
+Events are created when a new Processing Block or Scheduling Block Instance
+is created, cancelled etc.
+
+A higher level events API is also provided in the modules `pb_events.py` and
+`sbi_events.py`.
+
+"""
+from typing import List, Callable
+import ast
+
+from . import event_keys as keys
+from .config_db_redis import ConfigDb
+
+DB = ConfigDb()
+
+
+class Event:
+    """Event class."""
+
+    def __init__(self, event_id, aggregate_type, subscriber, event_data):
+        """Initialise the event."""
+        self.aggregate_type = aggregate_type
+        self.subscriber = subscriber
+        self._id = event_id
+        self._data = event_data
+
+    def __str__(self):
+        """Generate the 'informal' string representation.
+
+        Used by the print statement.
+        """
+        return str(dict(id=self._id, data=self._data))
+
+    def __repr__(self):
+        """Generate the 'official' string representation.
+
+        eg. used when printing lists of objects.
+        """
+        return '{}'.format(self._id)
+
+    @property
```

Comment:
Consider adding the the line:

`# pylint: disable=invalid-name`

on the line above `@property` to disable pylint comments about the function name `id` not conforming to style guidelines.

Reply:
this is a quite nice trick.